### PR TITLE
Use fetch() function that responds to the hard-coded JSON data and add Project ID for live server

### PR DESCRIPTION
**Pull request by jonaner**

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -35,7 +41,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
-          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
+          <deploy.projectId>jonathananer-step-2020</deploy.projectId>
```
Comment (ginellegaisano) [inline portfolio/pom.xml:44]: This change should also be accounted for in the PR description.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,10 +24,24 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private final ArrayList<String> messages = new ArrayList<String>();
```
Comment (jbir789) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: nit: you can replace `new ArrayList<String>()` with `new ArrayList<>`, as the type param is automatically inferred from the declaration.

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -1,5 +1,5 @@
 body {
-    background-color:rgb(198, 245, 245);
+    #background-color:rgb(198, 245, 245);
```
Comment (jbir789) [inline portfolio/src/main/webapp/style.css:2]: Is this change intended?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -23,10 +24,24 @@
...
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Jonathan");
+    messages.add("Something crazy");
+    messages.add("Something simple");
+    messages.add("Something amazing");
+    String json = "{";
```
Comment (ginellegaisano) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Alternatively, for creating the json, you could use String.format: https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#format(java.lang.String,%20java.lang.Object...)

Example:
String.format("My name is %s", name);

Comment (jonaner) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: That is a good suggestion.

Comment (jbir789) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Why not using Gson? It would automatically convert your list to a json String, which is a bit more scalable as you add more messages: https://github.com/google/gson

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -27,8 +27,22 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-function addGreeting(){
-    fetch('/data').then(response => response.text()).then((quote) => {
-      document.getElementById('quote-container').innerText = quote;
+function addMessage() {
+  fetch('/data').then(response => response.json()).then((message) => {
+    const messageList = document.getElementById('message-container');
+    messageList.innerHTML ='';
```
Comment (jbir789) [inline portfolio/src/main/webapp/script.js:33]: nit: add a space after the `=`

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -27,8 +27,22 @@ function addRandomGreeting() {
...
+    messageList.appendChild(
```
Comment (jbir789) [inline portfolio/src/main/webapp/script.js:34]: Would it be possible to write a loop that goes over the messages and automatically create a list element?

The idea would be that if we add more messages, we wouldn't need to change the code

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (jbir789) [reply portfolio/src/main/webapp/style.css:2]: I am not sure why we added the hashtag before background-color, could you clarify?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (jonaner) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: I believe that is next step, but I might be wrong.

Comment (jonaner) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Gson seems like a good way for the code to not just apply to one set messages. I've never used it before though.

Comment (jbir789) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Once you've added the dependency (which is explained in the week 3 walkthrough), you can do:

```java
private final Gson gson = new Gson();


private void myFunction() {
  String json = gson.toJson(messageList);
}
```

This will convert your array / list to a json String automatically.

Let me know if you need more help, happy to provide more details. This [page](https://www.tutorialspoint.com/gson/gson_quick_guide.htm) might also be useful to see an example on how Gson is setup. Note that they use a builder, which I don't think we have to worry at this time, so you can just instantiate Gson right away.

Comment (jonaner) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Okay thank you.

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (jbir789) [reply portfolio/src/main/webapp/script.js:34]: With the current json object, it might be difficult to implement a loop because the json isn't an array (it's an object with 3 fields, named `message1`, `message2` and `message3`) 

If you use Gson, it would instead convert your list into a Json array, and I believe you'd then be able to loop doing:

```javascript
for (var i = 0; i < message.length; i++) {
  messageList.appendChild(message[i]);
}
```

Comment (jonaner) [reply portfolio/src/main/webapp/script.js:34]: Okay, thank you for the clarification.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -23,10 +24,24 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private final ArrayList<String> messages = new ArrayList<>();
```
Comment (jbir789) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: nit: Use the interface type for the field: `private final List<String>`

If you want, you can also make it an ImmutableList, since it's not mutable at this time (optional though):

```java
private final ImmutableList<String> messages = ImmutableList.of("Something crazy", "Something simple", "Something amazing");
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (jbir789) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:28]: Can you still do this in this PR?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -15,18 +15,33 @@
...
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson; 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private final ArrayList<String> messages = new ArrayList<String>();
+  Gson gson = new Gson();
```
Comment (jbir789) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: It looks like there was a conflict here, I see 2 blocks of the same code?

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -27,8 +27,22 @@ function addRandomGreeting() {
...
-function addGreeting(){
-    fetch('/data').then(response => response.text()).then((quote) => {
-      document.getElementById('quote-container').innerText = quote;
+function addMessage() {
+  fetch('/data').then(response => response.json()).then((message) => {
+    
+    const messageList = document.getElementById('message-container');
+    messageList.innerHTML = '';
+    console.log(message[i]);
+    for (var i =0; i < message.length; i++){
```
Comment (jbir789) [inline portfolio/src/main/webapp/script.js:36]: Don't forget to remove the loop if we do this in a follow up PR
